src/beast/beast/http/message.h

```diff
@@ -96,271 +105,107 @@ struct message
-
-template<class Body, class Allocator = std::allocator<char>>
-using parsed_response = parsed_message<false, Body, Allocator>;
-
-template<
-    class RespBody, class RespAllocator,
-    class ReqBody, class ReqAllocator>
-response<RespBody, RespAllocator>&
-set_keep_alive(bool keep_alive,
-    response<RespBody, RespAllocator>& resp,
-        parsed_request<ReqBody, ReqAllocator> const& req)
-{
-    keep_alive = keep_alive && req.keep_alive;
-    if(keep_alive)
-    {
-        if(req.version < 11)
-            resp.headers.replace("Connection", "Keep-Alive");
-        else
-            resp.headers.erase("Connection");
-    }
-    else
-    {
-        if(req.version >= 11)
-            resp.headers.replace("Connection", "Close");
-        else
-            resp.headers.erase("Connection");
-    }
-    return resp;
-}
-
-//------------------------------------------------------------------------------
-
-enum connection_value
-{
-    close,
-    keep_alive,
-    upgrade
-};
-
-struct connection
-{
-    connection_value value;
-
-    connection(connection_value value_)
-        : value(value_)
-    {
-    }
-
-    /// Convert bool to keep-alive flag
-    connection(bool keep_alive)
-        : value(keep_alive ?
-            connection_value::keep_alive :
-                connection_value::close)
-    {
-    }
-};
-
-/** A prepared HTTP message. */
-template<bool isRequest, class Body, class Allocator>
-struct prepared_message : public message<isRequest, Body, Allocator>
-{
-    bool keep_alive;
-    bool upgrade;
-
-    prepared_message() = default;
-    prepared_message(prepared_message&&) = default;
-    prepared_message(prepared_message const&) = default;
-    prepared_message& operator=(prepared_message&&) = default;
-    prepared_message& operator=(prepared_message const&) = default;
-
-    /** Move-construct a prepared request.
+    choke();
 
-        @param msg The request to prepare.
-    */
-    template<class... Opts>
-    explicit
-    prepared_message(
-        request<Body, Allocator>&& msg,
-            Opts&&... opts);
-
-    /** Copy-construct a prepared request.
-
-        @param msg The request to prepare.
-    */
-    template<class... Opts>
-    explicit
-    prepared_message(
-        request<Body, Allocator> const& msg,
-            Opts&&... opts);
-
-    /** Move-construct a prepared response.
+    /// Wipe it
+    template<bool isRequest, class Body, class Headers>
+    friend 
+    void
+    choke(int);
 
-        @param msg The response to prepare.
+    /// Wipe it
+    template<bool isRequest, class Body, class Headers>
+    friend 
+    void
+    choke(double);
 
-        @param req The request we are preparing the response for.
-    */
-    template<class ReqBody, class ReqAllocator, class... Opts>
-    prepared_message(response<Body, Allocator>&& msg,
-        parsed_request<ReqBody, ReqAllocator> const& req,
-            Opts&&... opts);
+    void
+    droke();
```

Comment:
I'm not sure what's going on with `droke` and friends. I don't see a definition for them on the tip. Leftover code?